dotnet/src/SemanticKernel.Core/Data/TextSearchBehavior/TextSearchBehaviorOptions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Data;
+
+/// <summary>
+/// Contains options for the <see cref="TextSearchBehavior"/>.
+/// </summary>
+[Experimental("SKEXP0130")]
+public sealed class TextSearchBehaviorOptions
+{
+    private int _top = 3;
+
+    /// <summary>
+    /// Maximum number of results to return from the similarity search.
+    /// </summary>
+    /// <remarks>The value must be greater than 0.</remarks>
+    /// <value>The default value is 3 if not set.</value>
+    public int Top
+    {
+        get => this._top;
+        init
+        {
+            if (value < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), "Top must be greater than 0.");
+            }
+
```

Comment:
I would remove any validation from our side for this, as a good practice we should avoid to throw exceptions on property getter and setters.

```suggestion
            if (value < 1)
            {
                throw new ArgumentOutOfRangeException(nameof(value), "Top must be greater than 0.");
            }
```

Reply:
Why is it bad practice to throw ArgumentOutOfRange on a property?